diffrax/_solver/foster_langevin_srk.py

```diff
@@ -0,0 +1,410 @@
+    assert isinstance(diffusion, UnderdampedLangevinDiffusionTerm)
+    gamma = drift.gamma
+    u = drift.u
+    f = drift.grad_f
+    return gamma, u, f
+
+
+def _broadcast_pytree(source, target_tree):
+    # Broadcasts the source PyTree to the shape and PyTree structure of
+    # target_tree_shape. Requires that source is a prefix tree of target_tree
+    # This is used to case gamma and u to the shape of x0 and v0
+    def _inner_broadcast(_src_arr, _inner_target_tree):
+        _arr = jnp.asarray(_src_arr)
+
+        def fun(_leaf):
+            return jnp.asarray(
+                jnp.broadcast_to(_arr, _leaf.shape), dtype=jnp.result_type(_leaf)
+            )
+
+        return jtu.tree_map(fun, _inner_target_tree)
+
+    return jtu.tree_map(_inner_broadcast, source, target_tree)
+
+
+# CONCERNING COEFFICIENTS:
+# The coefficients used in a step of this SRK depend on
+# the time increment h, and the parameter gamma.
+# Assuming the modelled SDE stays the same (i.e. gamma is fixed),
+# then these coefficients must be recomputed each time h changes.
+# Furthermore, for very small h, directly computing the coefficients
+# via the function below can cause large floating point errors.
+# Hence, we pre-compute the Taylor expansion of the SRK coefficients
+# around h=0. Then we can compute the SRK coefficients either via
+# the Taylor expansion, or via direct computation.
+# In short the Taylor coefficients give a Taylor expansion with which
+# one can compute the SRK coefficients more precisely for a small h.
+
+
+class AbstractCoeffs(eqx.Module):
+    dtype: AbstractVar[jnp.dtype]
+
+
+_Coeffs = TypeVar("_Coeffs", bound=AbstractCoeffs)
+
+
+class SolverState(eqx.Module, Generic[_Coeffs]):
+    gamma: UnderdampedLangevinX
+    u: UnderdampedLangevinX
+    h: RealScalarLike
+    taylor_coeffs: PyTree[_Coeffs, "UnderdampedLangevinX"]
+    coeffs: _Coeffs
+    rho: UnderdampedLangevinX
+    prev_f: UnderdampedLangevinX
+
+
+class AbstractFosterLangevinSRK(
+    AbstractStratonovichSolver[SolverState],
+    AbstractItoSolver[SolverState],
+    Generic[_Coeffs, _ErrorEstimate],
+):
+    r"""Abstract class for Stochastic Runge Kutta methods specifically designed
+    for Underdamped Langevin Diffusion of the form
+
+    \begin{align*}
+        \mathrm{d} x(t) &= v(t) \, \mathrm{d}t \\
+        \mathrm{d} v(t) &= - \gamma \, v(t) \, \mathrm{d}t - u \,
+        \nabla \! f( x(t) ) \, \mathrm{d}t + \sqrt{2 \gamma u} \, \mathrm{d} w(t),
+    \end{align*}
+
+    where $x(t), v(t) \in \mathbb{R}^d$ represent the position
+    and velocity, $w$ is a Brownian motion in $\mathbb{R}^d$,
+    $f: \mathbb{R}^d \rightarrow \mathbb{R}$ is a potential function, and
+    $\gamma , u \in \mathbb{R}^{d \times d}$ are diagonal matrices governing
+    the friction and the inertia of the system.
+
+    Solvers which inherit from this class include [`diffrax.ALIGN`][],
+    [`diffrax.ShOULD`][], and [`diffrax.QUIC_SORT`][].
```

Comment:
Nit: spurious underscore in `QUICSORT`.

(Try building the documentation with `mkdocs serve` -- it should probably detect this and error out.)

Reply:
This docstring isn't included in the docs, so it didn't throw an error. But thanks for spotting it.